Add new terms for glossary: RHEL software management (#261)
redhat-documentation/supplementary-style-guide: supplementary_style_guide/glossary_terms_conventions/general_conventions/r.adoc

*See also*:

[[raw-interpreted-program]]
==== image:images/yes.png[yes] raw-interpreted program (noun)

> *Description*: _Raw-interpreted programs_ are programs that do not need to be compiled and are directly run by a language interpreter.

Reviewer: thinking a lot now about keeping the term singular in all definitions...or making them plural, as long as we do the same thing each time. if we go singular (which i think is more intuitive), change this one thusly:

A _raw-interpreted program_ is a program that doesn't [/does not] need to be..../?

*See also*:

[discrete]
[[raw-interpreted-program]]
==== image:images/yes.png[yes] raw-interpreted program (noun)

> *Description*: _Raw-interpreted language programs_ are programs that do not need to be compiled and are directly executed by a language interpreter.

Reviewer 2: Should "language" be removed from this, since the term doesn't have it?

```suggestion
*Description*: _Raw-interpreted programs_ are programs that do not need to be compiled and are directly executed by a language interpreter.
```

Author: +1

*See also*:

[[rolling-stream]]
==== image:images/yes.png[yes] Rolling Stream (noun)

> *Description*: _Rolling Streams_, also referred to as Rolling Application Streams, include Red Hat Enterprise Linux tools and applications that are updated frequently. Later versions of these streams replace earlier versions. Therefore, only one version at a time is supported.

Reviewer: ditto my comment above about defining a singular term in plural form. again, if we choose singular, change:

A _Rolling Stream_, also referred to as a Rolling Application Stream, includes..../?

*See also*:

> [[rolling-stream]]

Reviewer 3: I don't see the [discrete] tab. Is it still missing?

==== image:images/yes.png[yes] Rolling stream (noun)

> *Description*: _Rolling Application Streams_ include RHEL tools and applications that are updated frequently. Later versions of these streams replace earlier versions. Therefore, only one version is supported at a time.

Reviewer 2: Should "Application" be removed to be consistent with the term? or should it be included in the term? Also the capitalization is different for "Stream" - not sure what is right

Author: Thank you, I checked with the sst writer that the terms are interchangeable. Rewording this a bit.

> *Description*: _Rolling Application Streams_ include RHEL tools and applications that are updated frequently. Later versions of these streams replace earlier versions. Therefore, only one version is supported at a time.

Reviewer 4: ```suggestion
*Description*: _Rolling Application Streams_ include Red Hat Enterprise Linux tools and applications that are updated frequently. Later versions of these streams replace earlier versions. Therefore, only one version is supported at a time.
```
- Spell out RHEL as you do elsewhere in this PR?

> *Description*: _Rolling Application Streams_ include Red Hat Enterprise Linux tools and applications that are updated frequently. Later versions of these streams replace earlier versions. Therefore, only one version at a time is supported.

Reviewer: term is "Rolling stream," but description says "Rolling [App] Streams"

- additional word ("[App]")
- cap S versus lc s in term

clarify?

Author: Confirmed with SMEs, adding capitalization

> *Description*: _Rolling Application Streams_ include RHEL tools and applications that move very fast and are updated frequently. Later versions of these streams replace versions released earlier. Therefore, only one version is supported at a time.

Reviewer 4: ```suggestion
*Description*: _Rolling Application Streams_ include RHEL tools and applications that are updated frequently. Later versions of these streams replace versions released earlier. Therefore, only one version is supported at a time.
```
- Omit `move very fast` because tools and applications can't actually move.

[[rpm]]
==== image:images/yes.png[yes] RPM (noun)

> *Description*: _RPM_ is the recursive initialism for the _RPM Package Manager_. RPM manages files in the RPM format. Those files are referred to as "RPM packages". Even though RPM packages are known informally as `rpm` files, do not use this informal term in Red Hat documentation to avoid confusion with the command name.

Reviewer: i'm sure i'm not the only one of us who's bothered by "RPM" standing for "RPM Package Manager," bc of the obvious redundancy. worth it to add "Originally known as Red Hat Package Manager" somewhere?

also, consider this change (for easier readability): "...Even though..., to avoid confusion with the command name, do not use this informal...."/?

Author: RPM already discussed in one of the comments above

*Use it*: yes

*Incorrect forms*: rpm

*See also*:

[[rpm-macro]]
==== image:images/yes.png[yes] RPM macro (noun)
*Description*: An _RPM macro_ is a straight text substitution that can be conditionally assigned based on the optional evaluation of a statement when certain built-in functionality is used. RPM macros help to make package maintenance simpler and to automate certain repetitive tasks or patterns that are common across all types of packages.

*Use it*: yes

> *Incorrect forms*:

Reviewer: add "rpm macro" or obvious?

[discrete]
[[rpm]]
==== image:images/yes.png[yes] RPM (noun)

> *Description*: _RPM_ is the recursive initialism for the _RPM Package Manager_. RPM manages files in the RPM format. Those files are referred to as "RPM packages". Note that even though RPM packages are known informally as `rpm` files, this informal term must not be used in Red Hat documentation to avoid confusion with the command name.

Reviewer 3: Doesn't 'RPM' stand for 'Red Hat Package Manager'? You all know more about this than I do, but I am doubly confused after reading the description.

Reviewer 4: @IngridT1 - You would've been correct before 17 Jan 2000: https://rpm.org/timeline.html. That's when they changed the meaning from "Red Hat Package Manager" to a recursive acronyn of "RPM Package Manager".

> *Description*: _RPM_ is the recursive acronym for the _RPM Package Manager_. RPM manages files in the RPM format, known as RPM packages. RPM packages are known informally as rpm files, but this informal usage is not used in Red Hat documentation to avoid confusion with the command name. Files in RPM format are referred to as RPM packages.

Reviewer 2: * Initialism was actually the correct term here
* I think the quotes you removed were correct
* If the backticks are being removed from `rpm`, then I think it needs to be capitalized as RPM files

Author: Thank you for catching this. Applying the feedback and rewording a bit.

*Use it*: yes

*Incorrect forms*: rpm

*See also*:

[discrete]
[[rpm-macro]]
==== image:images/yes.png[yes] RPM macro (noun)

> *Description*: An _RPM macro_ is a straight text substitution that can be conditionally assigned based on the optional evaluation of a statement when certain built-in functionality is used. RPM macros help to make package maintenance simpler and to automate certain repetitive tasks or patterns that are common across all types of packages. An RPM macro is a fragment of text with an assigned name. Whenever the name is used, it is replaced by the expanded body of the macro. This process is called macro expansion, and it can range from a static text substitution to an arbitrary code execution with side effects.

Reviewer 4: ```suggestion
*Description*: An _RPM macro_ is a straight text substitution that can be conditionally assigned based on the optional evaluation of a statement when certain built-in functionality is used. RPM macros help to make package maintenance simpler and to automate certain repetitive tasks or patterns that are common across all types of packages. An RPM macro is a fragment of text with an assigned name. Whenever the name is used, it is replaced by the expanded body of the macro. This process is called "macro expansion", and it can range from a static text substitution to an arbitrary code execution with side effects.
```

Author: Thank you. Checking on the term with the writer; could be added to the next batch.

> *Description*: An _RPM macro_ is a straight text substitution that can be conditionally assigned based on the optional evaluation of a statement when certain built-in functionality is used. RPM macros help to make package maintenance simpler and to automate certain repetitive tasks or patterns that are common across all types of packages.  An RPM macro is a fragment of text with an assigned name. Whenever the name is used, it is replaced by the expanded body of the macro. This process is called macro expansion, and it can range from a static text substitution to an arbitrary code execution with side effects.

Reviewer 4: ```suggestion
*Description*: An _RPM macro_ is a straight text substitution that can be conditionally assigned based on the optional evaluation of a statement when certain built-in functionality is used. RPM macros help to make package maintenance simpler and to automate certain repetitive tasks or patterns that are common across all types of packages. An RPM macro is a fragment of text with an assigned name. Whenever the name is used, it is replaced by the expanded body of the macro. This process is called macro expansion, and it can range from a static text substitution to an arbitrary code execution with side effects.
```
- omit extra space

Author: Good catch, thank you!

[discrete]
[[rpm]]
==== image:images/yes.png[yes] RPM (noun)
*Description*: _RPM_ is the recursive initialism for the _RPM Package Manager_. RPM manages files in the RPM format. Those files are referred to as "RPM packages". Even though RPM packages are known informally as `rpm` files, do not use this informal term in Red Hat documentation to avoid confusion with the command name.

*Use it*: yes

*Incorrect forms*: rpm

*See also*:

[discrete]
[[rpm-macro]]
==== image:images/yes.png[yes] RPM macro (noun)

> *Description*: An _RPM macro_ is a straight text substitution that can be conditionally assigned based on the optional evaluation of a statement when certain built-in functionality is used. RPM macros help to make package maintenance simpler and to automate certain repetitive tasks or patterns that are common across all types of packages. An RPM macro is a fragment of text with an assigned name. Whenever the name is used, it is replaced by the expanded body of the macro. This process is called "macro expansion" and can range from a static text substitution to arbitrary code running with side effects.

Reviewer 5: This is a very long definition. Can we cut the last two sentences ("An RPM macro is a fragment... running with side effects")?

Author: It  is actually 3 sentences :-) discussed with SME and cutting those sentences.
